paddle/fluid/pybind/pybind.cc

```diff
@@ -1221,11 +1222,39 @@ All parameter, weight, gradient are variables in Paddle.
            const std::unordered_set<std::string> &no_grad_set,
            const std::vector<BlockDesc *> &grad_sub_block) {
           std::unordered_map<std::string, std::string> grad_to_var;
-          std::vector<std::unique_ptr<OpDesc>> grad_op_descs =
-              framework::OpInfoMap::Instance()
-                  .Get(op_desc.Type())
-                  .GradOpMaker()(
-                      op_desc, no_grad_set, &grad_to_var, grad_sub_block);
+
+          auto op_info = framework::OpInfoMap::Instance().Get(op_desc.Type());
+          std::vector<std::unique_ptr<OpDesc>> grad_op_descs;
+
+          // In PrimEnabled mode, the priority of GradCompOpMaker is grater than
+          // GradCompMaker as we need split first-order grad operator into
+          // primitive operators for compiler. In PrimDisabled mode, the
+          // priority of GradCompOpMaker is less than GradCompMaker for better
+          // performance.
+          if (paddle::prim::PrimCommonUtils::IsPrimEnabled()) {
+            if (op_info.GradCompOpMaker() != nullptr) {
+              grad_op_descs = op_info.GradCompOpMaker()(op_desc,
+                                                        no_grad_set,
+                                                        &grad_to_var,
+                                                        op_desc.Block(),
+                                                        grad_sub_block);
+            } else {
+              grad_op_descs = op_info.GradOpMaker()(
+                  op_desc, no_grad_set, &grad_to_var, grad_sub_block);
```

Comment:
Will this throw error when we can't find neither CompGradMaker nor GradMaker?

Reply:
done